### PR TITLE
ICS20-new-folder-structure-for-version-support

### PR DESCRIPTION
**Pull request by sangier**
Objective: This PR introduces a new folder structure to accommodate multiple versions of the ICS20 specification. The update facilitates easier management and access to different versions of the documentation.

Changes Made:

- Created a new directory structure to support versioning of the ICS20 spec.
- The readme.md files have been organized into version-specific folders to ensure clarity and maintain historical integrity of the document evolution.

New Structure Details:
- ics20/readme.md: This file now represents the latest version of the specification (v2). It has been updated to reflect the most current information and standards.
- ics20/v1/readme.md: This file has been restored to its state from a previous version, specifically [commit 0238989da88e790ec4daea7214ff3521cc264137](https://github.com/cosmos/ibc/commit/0238989da88e790ec4daea7214ff3521cc264137). It represents the v1 version of the ICS20 spec.

Purpose:
The new folder structure and version-specific documentation are designed to:

Improve navigation and accessibility for users looking to find specific versions of the ICS20 spec.
Maintain a clear and traceable history of changes and updates made over time.

### CODE REVIEW

**Comment by crodriguezvega** [comment]
@sangier it seems like the link to [commit 0238989da88e790ec4daea7214ff3521cc264137](https://github.com/your-repository-link/commit/0238989da88e790ec4daea7214ff3521cc264137) is not working.

**Comment by sangier** [comment]
@crodriguezvega description commit link updated!

File: spec/app/ics-020-fungible-token-transfer/README.md [hunk 1]
```diff
@@ -792,9 +793,9 @@ July 27, 2020 - Re-addition of source field
 
 Nov 11, 2022 - Addition of a memo field
 
-Sep 22, 2023 - Support for multi-token packets
+Sep 22, 2023 - Support for multi-token packets - [PR](https://github.com/cosmos/ibc/pull/1020)
```
Comment (crodriguezvega) [inline spec/app/ics-020-fungible-token-transfer/README.md:796]:
```suggestion
Sep 22, 2023 - [Support for multi-token packets](https://github.com/cosmos/ibc/pull/1020)
```

File: spec/app/ics-020-fungible-token-transfer/README.md [hunk 2]
```diff
@@ -792,9 +793,9 @@ July 27, 2020 - Re-addition of source field
...
 
-March 5, 2024 - Support for ics20-2
+March 5, 2024 - Support for ics20-2 - [PR](https://github.com/cosmos/ibc/pull/1090)
```
Comment (crodriguezvega) [inline spec/app/ics-020-fungible-token-transfer/README.md:798]:
```suggestion
March 5, 2024 - [Support for path forwarding](https://github.com/cosmos/ibc/pull/1090)
```

File: spec/app/ics-020-fungible-token-transfer/README.md [hunk 3]
```diff
@@ -792,9 +793,9 @@ July 27, 2020 - Re-addition of source field
...
```
Comment (crodriguezvega) [inline spec/app/ics-020-fungible-token-transfer/README.md:799]: Even though the [PR to use protobuf encoding](https://github.com/cosmos/ibc/pull/1118) was merged into the path forwarding branch, maybe it's worth to add an entry for it in the history section.

**Review by crodriguezvega — APPROVED** [review]
Thank you so much for doing this, @sangier! I just left a few small comments

File: spec/app/ics-020-fungible-token-transfer/README.md (hunk 3 above)
Comment (sangier) [reply spec/app/ics-020-fungible-token-transfer/README.md:801]: Agreed and [applied](https://github.com/cosmos/ibc/pull/1124/commits/e5b9e733a34c6fc2df83bad63e7f1b7641fe9e84)!

File: spec/app/ics-020-fungible-token-transfer/README.md [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 stage: draft
 category: IBC/APP
 requires: 25, 26
 kind: instantiation
 version compatibility: ibc-go v7.0.0 (ics20-1 supported only)
-author: Christopher Goes <cwgoes@interchain.berlin>
+author: Christopher Goes <cwgoes@interchain.berlin>, Aditya Sripal <aditya@interchain.io>,
 created: 2019-07-15 
 modified: 2024-03-05
 ---
```
Comment (AdityaSripal) [inline spec/app/ics-020-fungible-token-transfer/README.md:13]: Maybe we should have a note here or at the top which ibc-go/ibc-rs versions support v1 and which support v2

**Comment by sangier** [comment]
Mmm, I am wondering if we should keep track of the different spec versions in the [main page](https://github.com/cosmos/ibc?tab=readme-ov-file#app). With the current structure and with this modification, the new version would be linked but the stage and implementation parameters will be related to the old version. We can either modify the stage and implementation details, or create a new entry with the new version details. What do you prefer?

**Comment by crodriguezvega** [comment]
I am going to merge this now because I want to add the mermaid diagrams, and doing it on top of this PR will prevent merge conflicts later on.

**Pull request merged**